contracts/Memeketplace.sol

```diff
@@ -29,20 +33,44 @@ contract MemeketPlace {
     }
 
     function likeMeme(uint256 _memeId) public {
+        require(likes[_memeId][msg.sender] != 1, "You have already liked this meme");
+        likes[_memeId][msg.sender] == 1;
         memeContract.setMemeLikes(
             _memeId,
             memeContract.getMemeLikes(_memeId).add(1)
         );
     }
 
     function dislikeMeme(uint256 _memeId) public {
+        require(likes[_memeId][msg.sender] != 2, "You have already disliked this meme");
```

Comment:
@TzeMeng can I check with you if this checks if user already liked the meme, then when he dislikes the meme, there will be `memeContract.setMemeLikes(_memeId).sub(1) `?

Reply:
I think the sequence would be removeLike -> dislike, den if user only remove his like without disliking the meme it would just be removeLike

Reply:
I think to reduce less clicks, user can dislike a meme even after liking a meme, instead of having to remove the like first , then dislike. 
As well, should the mappings in line 12 be more suited for `memeContract`?
`mapping(address => bool) public hasUserLike`
`mapping(address => bool) public hasUserDislike`
`mapping(address => bool) public hasUserFlag`

when user clicks like : 
1. check `hasUserLike `(has it been liked already) -> if yes, then` setMemeLike.sub(1),` then `setMemeLike.add(1)` else : 
2. check `hasUserDislike` (see if it has been disliked before) -> if yes, then minus 1 in `setMemeDislike.sub(1),` then` setMemeLike.add(1)` else:
3. `setMemLike.add(1)`